### PR TITLE
Allow disable outbound snat when Azure standard load balancer is used

### LINKED ISSUE #75281: Azure outbound rules is not taking effect when using together with standard loadbalancer
**What happened**:

[Outbound rules](https://docs.microsoft.com/en-us/azure/load-balancer/load-balancer-outbound-rules-overview) is the recommended way to configure the outbound networks together with Azure standard loadbalancer (SLB).

However, `disableOutboundSNAT` for load balancer rules are disabled by default. That means all frontend IPs would be used as the SNAT IPs for virtual machines, according to [Azure docs](https://docs.microsoft.com/en-us/azure/load-balancer/load-balancer-outbound-connections#standard-load-balancer). This would introduce troubles for various security scenarios, as there's no way to get a stable list of egress IPs.

Azure cloud provider should support this scenario, and disable the outbound SNAT when outbound rules are configured by provisioning tools.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/kind bug
/sig azure
/priority critical-urgent

### PR DESCRIPTION
**Pull request by feiskyer**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind bug

**What this PR does / why we need it**:

Azure outbound rules are not taking effect when used together with standard load balancer .

This PR adds a new configuration for disabling SNAT for load balancer rules, so that outbound rules would be used as the egress IPs.

In such case, disableOutboundSNAT should be set to true and loadBalancerSku should be set to standard.

**Which issue(s) this PR fixes**:

Fixes #75281

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Allow disable outbound SNAT when Azure standard load balancer is used together with outbound rules. 
```

### CODE REVIEW

**Comment by feiskyer** [comment]
Added to v1.14 since it is a critical bug fix.

/kind bug
/sig azure
/milestone v1.14
/priority critical-urgent

File: pkg/cloudprovider/providers/azure/azure.go [hunk 1]
```diff
@@ -321,9 +326,20 @@ func NewCloud(configReader io.Reader) (cloudprovider.Interface, error) {
 		config.CloudProviderBackoffDuration = backoffDurationDefault
 	}
 
-	// Do not add master nodes to standard LB by default.
-	if config.ExcludeMasterFromStandardLB == nil {
-		config.ExcludeMasterFromStandardLB = &defaultExcludeMasterFromStandardLB
+	if strings.EqualFold(config.LoadBalancerSku, loadBalancerSkuStandard) {
+		// Do not add master nodes to standard LB by default.
+		if config.ExcludeMasterFromStandardLB == nil {
+			config.ExcludeMasterFromStandardLB = &defaultExcludeMasterFromStandardLB
```
Comment (andyzhangx) [inline pkg/cloudprovider/providers/azure/azure.go]: this fixed another issue?

Comment (feiskyer) [reply pkg/cloudprovider/providers/azure/azure.go]: No, it's just moving the original condition within standard sku. ExcludeMasterFromStandardLB is only applied for standard LB.

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**